openedx/core/djangoapps/courseware_api/serializers.py

```diff
@@ -56,6 +56,30 @@ class Meta:
         ref_name = 'courseware_api'
 
 
+class CourseProgramSerializer(serializers.Serializer):
+    progress = serializers.SerializerMethodField()
+    title = serializers.CharField()
+    type = serializers.CharField()
+    url = serializers.SerializerMethodField()
+    uuid = serializers.SerializerMethodField()
+
+    def get_progress(self, program):
+        progress = program.get('progress')
```

Comment:
So you're using `.get` here which means it will default to `None` if it isn't found. Is that a concern with this code (as in could that happen or will progress always exist)? If it isn't guaranteed to exist, then you'll end up throwing an error below when you call `progress.get()` because in effect, you're saying `None.get()`

One solution would be to supply a default value here to avoid any other errors.

Looking below at the rest of this code, you use `.get` often, but don't handle the case where it might not exist. Something to think about moving forward.

Also, you might want to look up the `source` field for the uuid. I think you might be able to do something like `uuid = serializers.UUIDField(source='progress.uuid')` or something like that.

Reply:
Progress (along with `completed`, `not_started`, and `in_progress`) will always exist here because it's being populated and passed through the view.